Components/ProductToShop.php

```diff
@@ -445,6 +449,25 @@ private function deleteDetail(DetailModel $detailModel)
         }
     }
 
+    /**
+     * @param DetailModel $detailModel
+     * @param $article
+     */
+    private function removeOptions(DetailModel $detailModel, $article)
+    {
+        $configSet = $article->getConfiguratorSet();
+        $options = $detailModel->getConfiguratorOptions();
+        if($options && $configSet) {
+            $optionCollection = $configSet->getOptions();
+            /** @var Option $option */
+            foreach ($options as $option) {
+                $optionCollection->removeElement($option);
+            }
+            $configSet->setOptions($options);
```

Comment:
does `setOptions()` append or does it overwrite the list of options?
i think it overrides and then the whole foreach is not necessary